diff --git a/src/NETGENPlugin/NETGENPlugin_Mesher.cxx b/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
--- a/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
@@ -41,14 +41,14 @@
 #include <SMESH_ComputeError.hxx>
 #include <SMESH_ControlPnt.hxx>
 #include <SMESH_File.hxx>
-#include <SMESH_Gen_i.hxx>
+#include <SMESH_Gen.hxx>
 #include <SMESH_Mesh.hxx>
 #include <SMESH_MesherHelper.hxx>
 #include <SMESH_subMesh.hxx>
 #include <StdMeshers_QuadToTriaAdaptor.hxx>
 #include <StdMeshers_ViscousLayers2D.hxx>

-#include <SALOMEDS_Tool.hxx>
+// #include <SALOMEDS_Tool.hxx>

 #include <utilities.h>

@@ -647,6 +647,7 @@
     // const char*
     mparams.meshsizefilename= hyp->GetMeshSizeFile().empty() ? 0 : hyp->GetMeshSizeFile().c_str();
 #endif
+    /*
     const NETGENPlugin_Hypothesis::TLocalSize& localSizes = hyp->GetLocalSizesAndEntries();
     if ( !localSizes.empty() )
     {
@@ -668,6 +669,7 @@
         setLocalSize(S, val);
       }
     }
+    */
   }
 }

@@ -855,12 +857,12 @@
       {
         Standard_Integer n1,n2,n3;
         triangulation->Triangles()(i).Get( n1,n2,n3 );
-        p [0] = triangulation->Nodes()(n1).Transformed(loc).XYZ();
-        p [1] = triangulation->Nodes()(n2).Transformed(loc).XYZ();
-        p [2] = triangulation->Nodes()(n3).Transformed(loc).XYZ();
-        uv[0] = triangulation->UVNodes()(n1).XY();
-        uv[1] = triangulation->UVNodes()(n2).XY();
-        uv[2] = triangulation->UVNodes()(n3).XY();
+        p [0] = triangulation->Node(n1).Transformed(loc).XYZ();
+        p [1] = triangulation->Node(n2).Transformed(loc).XYZ();
+        p [2] = triangulation->Node(n3).Transformed(loc).XYZ();
+        uv[0] = triangulation->UVNode(n1).XY();
+        uv[1] = triangulation->UVNode(n2).XY();
+        uv[2] = triangulation->UVNode(n3).XY();
         surfProp.SetParameters( uv[0].X(), uv[0].Y() );
         if ( !surfProp.IsCurvatureDefined() )
           break;
@@ -1018,17 +1020,15 @@
       BRep_Tool::Triangulation ( TopoDS::Face( fExp.Current() ), loc);
     if ( triangulation.IsNull() ) continue;
     const double fTol = BRep_Tool::Tolerance( TopoDS::Face( fExp.Current() ));
-    const TColgp_Array1OfPnt&   points = triangulation->Nodes();
-    const Poly_Array1OfTriangle& trias = triangulation->Triangles();
-    for ( int iT = trias.Lower(); iT <= trias.Upper(); ++iT )
+    for ( int iT = 0; iT <= triangulation->NbTriangles(); ++iT )
     {
-      trias(iT).Get( i1, i2, i3 );
+      triangulation->Triangle(iT).Get( i1, i2, i3 );
       for ( int j = 0; j < 3; ++j )
       {
-        double dist2 = points(*pi[j]).SquareDistance( points( *pi[j+1] ));
+        double dist2 = triangulation->Node(*pi[j]).SquareDistance( triangulation->Node( *pi[j+1] ));
         if ( dist2 < minh && fTol*fTol < dist2 )
           minh = dist2;
-        bb.Add( points(*pi[j]));
+        bb.Add( triangulation->Node(*pi[j]));
       }
     }
   }

@@ -4475,7 +4477,8 @@

 std::string NETGENPlugin_NetgenLibWrapper::getOutputFileName()
 {
-  std::string aTmpDir = SALOMEDS_Tool::GetTmpDir();
+  // std::string aTmpDir = SALOMEDS_Tool::GetTmpDir();
+  std::string aTmpDir = "/tmp";

   TCollection_AsciiString aGenericName = aTmpDir.c_str();
   aGenericName += "NETGEN_";
@@ -4519,6 +4522,7 @@

 void NETGENPlugin_NetgenLibWrapper::removeOutputFile()
 {
+  /*
   if ( !_outputFileName.empty() )
   {
     if ( _ngcout )
@@ -4536,4 +4540,5 @@

     SALOMEDS_Tool::RemoveTemporaryFiles( tmpDir.c_str(), aFiles, true );
   }
+  */
 }
--- a/src/NETGENPlugin/NETGENPlugin_NETGEN_2D_ONLY.cxx	2021-10-19 16:56:46.435802635 -0400
+++ b/src/NETGENPlugin/NETGENPlugin_NETGEN_2D_ONLY.cxx	2021-12-23 15:50:52.845617816 -0500
@@ -64,6 +64,7 @@
 //#include <meshtype.hpp>
 namespace netgen {
   NETGENPLUGIN_DLL_HEADER
+  extern OCCParameters occparam;
   extern MeshingParameters mparam;
   extern void OCCSetLocalMeshSize(OCCGeometry & geom, Mesh & mesh);
 }
@@ -288,7 +288,7 @@
     //netgen::occparam.resthcloseedgefac = 1.0 + netgen::mparam.grading;
 #endif
     occgeoComm.face_maxh = netgen::mparam.maxh;
-    netgen::OCCSetLocalMeshSize( occgeoComm, *ngMeshes[0] );
+    netgen::OCCSetLocalMeshSize( occgeoComm, *ngMeshes[0], netgen::mparam, netgen::occparam );
     occgeoComm.emap.Clear();
     occgeoComm.vmap.Clear();

