--- a/src/StdMeshers/StdMeshers_ViscousLayers.cxx	2022-03-20 20:17:36.000942449 -0400
+++ b/src/StdMeshers/StdMeshers_ViscousLayers.cxx	2022-03-20 20:50:28.098015939 -0400
@@ -52,7 +52,6 @@
 #include "StdMeshers_Quadrangle_2D.hxx"
 #include "StdMeshers_ViscousLayers2D.hxx"
 
-#include <Adaptor3d_HSurface.hxx>
 #include <BRepAdaptor_Curve.hxx>
 #include <BRepAdaptor_Curve2d.hxx>
 #include <BRepAdaptor_Surface.hxx>
@@ -1818,8 +1817,8 @@
     //case GeomAbs_SurfaceOfExtrusion:
     case GeomAbs_OffsetSurface:
     {
-      Handle(Adaptor3d_HSurface) base = surface.BasisSurface();
-      return getRovolutionAxis( base->Surface(), axis );
+      Adaptor3d_Surface base = *surface.BasisSurface();
+      return getRovolutionAxis( base, axis );
     }
     default: return false;
     }
