diff -r a/src/StdMeshers/StdMeshers_ViscousLayers.cxx b/src/StdMeshers/StdMeshers_ViscousLayers.cxx
--- a/src/StdMeshers/StdMeshers_ViscousLayers.cxx	2021-10-19 16:58:36.742163609 -0400
+++ b/src/StdMeshers/StdMeshers_ViscousLayers.cxx	2021-12-23 10:16:07.654839133 -0500
@@ -51,7 +51,6 @@
 #include "StdMeshers_ProjectionUtils.hxx"
 #include "StdMeshers_ViscousLayers2D.hxx"

-#include <Adaptor3d_HSurface.hxx>
 #include <BRepAdaptor_Curve.hxx>
 #include <BRepAdaptor_Curve2d.hxx>
 #include <BRepAdaptor_Surface.hxx>
@@ -1751,8 +1750,8 @@
     //case GeomAbs_SurfaceOfExtrusion:
     case GeomAbs_OffsetSurface:
     {
-      Handle(Adaptor3d_HSurface) base = surface.BasisSurface();
-      return getRovolutionAxis( base->Surface(), axis );
+      Adaptor3d_Surface base = *surface.BasisSurface();
+      return getRovolutionAxis( base, axis );
     }
     default: return false;
     }
