diff -ru a/src/Controls/SMESH_Controls.cxx b/src/Controls/SMESH_Controls.cxx
--- a/src/Controls/SMESH_Controls.cxx
+++ b/src/Controls/SMESH_Controls.cxx
@@ -4361,6 +4361,8 @@
   void SetFlag  ( int flag ) { myFlags |= flag; }
   void UnsetFlag( int flag ) { myFlags &= ~flag; }

+  Classifier(Classifier&&) noexcept(false) = default;
+
 private:
   bool isOutOfSolid (const gp_Pnt& p);
   bool isOutOfBox   (const gp_Pnt& p);
